kvbc/src/ReplicaImp.cpp

```diff
@@ -128,14 +128,13 @@ Status ReplicaImp::getBlockData(BlockId blockId, SetOfKeyValuePairs &outBlockDat
   // TODO(GG): check legality of operation (the method should be invoked from
   // the replica's internal thread)
 
-  Sliver block = getBlockInternal(blockId);
-
-  if (block.length() == 0) {
+  try {
+    Sliver block = getBlockInternal(blockId);
+    outBlockData = block::getData(block);
+  } catch (const NotFoundException &e) {
```

Comment:
I am thinking if it will it be safe to throw any other exceptions out of this function. Most of the code doesn't expect exceptions and it might be safer to catch all exceptions here, at least for now. What do you think?

Reply:
In every other case the application has exited; now the runtime_exception will be caught at the topmost layer

Reply:
other exceptions here means problems with db/consistency